### PR TITLE
Bump env

### PR DESCRIPTION
**Pull request by sisuresh**
# Description

Bumps core to latest env and adapts for vnext and prev.

# Checklist
- [ ] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [ ] Rebased on top of master (no merge commits)
- [ ] Ran `clang-format` v8.0.0 (via `make format` or the Visual Studio extension)
- [ ] Compiles
- [ ] Ran all tests
- [ ] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

### CODE REVIEW

File: src/transactions/InvokeHostFunctionOpFrame.cpp [hunk 1]
```diff
@@ -69,8 +69,29 @@ getLedgerInfo(AbstractLedgerTxn& ltx, Application& app,
     info.min_temp_entry_ttl =
         sorobanConfig.stateArchivalSettings().minTemporaryTTL;
     info.max_entry_ttl = sorobanConfig.stateArchivalSettings().maxEntryTTL;
-    info.cpu_cost_params = toCxxBuf(sorobanConfig.cpuCostParams());
-    info.mem_cost_params = toCxxBuf(sorobanConfig.memCostParams());
+
+    auto cpu = sorobanConfig.cpuCostParams();
+    auto mem = sorobanConfig.memCostParams();
+
+// The host expects this vector to be the size of the # of cost types in the
```
Comment (dmkozh) [inline src/transactions/InvokeHostFunctionOpFrame.cpp:76]: This seems kind of hacky, maybe we should have protocol-based checks on the host side? I'm not sure we should be using new cost types in protocol 20.

Comment (sisuresh) [reply src/transactions/InvokeHostFunctionOpFrame.cpp:76]: Yeah I agree this is pretty hacky, but [cost_models](https://github.com/stellar/rs-soroban-env/blob/f8d3bafa19b934741efbb6f8ef3311696b3556de/soroban-env-host/src/budget/dimension.rs#L18C16-L18C27) is an array, so the length is set at compile time. We could dynamically allocate it, but I went with this path instead. I'm not opposed to fixing this in env though.

Comment (dmkozh) [reply src/transactions/InvokeHostFunctionOpFrame.cpp:76]: Yeah, I think it should probably be a vector. But I think we can try to refactor that as a followup after p21 is out.

File: src/testdata/ledger-close-meta-v1-protocol-21-soroban.json [hunk 2]
```diff
@@ -6,27 +6,27 @@
...
                         }
                     },
-                    "txSetResultHash": "ca00e9e245a55766694741fdc6e4f4237feb4cb16b7f5650b7d90aa676f39550",
-                    "bucketListHash": "874234dba835a0c540a614b637e9d7c19ccc4209d5fa97ec1bc048d241fc26e5",
+                    "txSetResultHash": "e97735cdc74c7a7c2ca6de341539aa7e7a37916158d8758a830c1c971ca16f3b",
+                    "bucketListHash": "0ae52c9b27859cba44d9467bd2d32726240bb7ea995956cd977cfacee6653819",
                     "ledgerSeq": 23,
                     "totalCoins": 1000000000000000000,
-                    "feePool": 384917,
+                    "feePool": 384918,
```
[the remark this answers is not in the document]
Comment (sisuresh) [reply src/testdata/ledger-close-meta-v1-protocol-21-soroban.json:29]: I assumed it was due to extendFootprintOp extending CONTRACT_CODE in one of the transactions, which is a little bigger due to the module cache update in v21.

File: src/ledger/NetworkConfig.cpp [hunk 3]
```diff
@@ -315,12 +322,111 @@ initialCpuCostParamsEntry()
...
+            break;
+        case InstantiateWasmImports:
+            params[val] = ContractCostParamEntry{ExtensionPoint{0}, 0, 828974};
+            break;
+        case InstantiateWasmExports:
+            params[val] = ContractCostParamEntry{ExtensionPoint{0}, 0, 297100};
+            break;
+        case InstantiateWasmDataSegmentBytes:
+            params[val] = ContractCostParamEntry{ExtensionPoint{0}, 0, 14};
+            break;
```
[the remark this answers is not in the document]
Comment (sisuresh) [reply src/ledger/NetworkConfig.cpp:421]: Good catch. Fixed

File: src/transactions/InvokeHostFunctionOpFrame.cpp (hunk 1 above)
Comment (sisuresh) [reply src/transactions/InvokeHostFunctionOpFrame.cpp:76]: https://github.com/stellar/rs-soroban-env/issues/1388

**Comment by dmkozh** [comment]
r+ bc205ca

**Pull request merged**